### PR TITLE
library: refact ceph_dashboard_user

### PR DESCRIPTION
**Pull request by guits**
refact this module due to recent changes in ceph pacific.
The password must be passed with `-i` option.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

File: module_utils/ca_common.py [hunk 1]
```diff
@@ -26,20 +26,24 @@ def generate_ceph_cmd(sub_cmd, args, user_key=None, cluster='ceph', user='client
...
-    command_exec = [container_binary,
-                    'run',
-                    '--rm',
-                    '--net=host',
-                    '-v', '/etc/ceph:/etc/ceph:z',
-                    '-v', '/var/lib/ceph/:/var/lib/ceph/:z',
-                    '-v', '/var/log/ceph/:/var/log/ceph/:z',
-                    '--entrypoint=' + binary, container_image]
+    command_exec = []
+    command_exec.extend([container_binary, 'run'])
```
Comment (dsavineau) [inline module_utils/ca_common.py:36]: I think you can simply do `command_exec = [container_binary, 'run']`

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 2]
```diff
@@ -150,7 +150,9 @@
   changed_when: false
 
 - name: set grafana api password
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-password {{ grafana_admin_password }}"
+  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-password -i -"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:153]: You need to also add the `--interactive` container option there I think (after `container_exec_cmd`)

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 3]
```diff
@@ -215,12 +217,16 @@
       changed_when: false
 
     - name: set the rgw access key
-      command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-rgw-api-access-key {{ rgw_access_key }}"
+      command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-rgw-api-access-key -i -"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:220]: ditto

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 4]
```diff
@@ -215,12 +217,16 @@
...
+      args:
+        stdin: "{{ rgw_access_key }}"
       delegate_to: "{{ groups[mon_group_name][0] }}"
       changed_when: false
 
     - name: set the rgw secret key
-      command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-rgw-api-secret-key {{ rgw_secret_key }}"
+      command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-rgw-api-secret-key -i -"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:227]: ditto

**Review by dsavineau — CHANGES_REQUESTED** [review]
if I remember correctly you also need to do the change for the ISCSI tasks (`iscsi-gateway-add`)

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test centos-non_container-purge

**Review by dsavineau — APPROVED** [review]

**Pull request merged**